### PR TITLE
Add tests for controllers

### LINKED ISSUE #137: Implementing tests for controllers and mailers using rspec
Tests will have to written using RSpec for controller and mailers. They will prove to be necessary while setting up a CI/CD pipeline. 
Refer - https://github.com/rspec/rspec-rails

### PR DESCRIPTION
**Pull request by ankitkataria**
Fixes #137 

#### Describe the changes you have made in this pr -
Initialized tests for controllers and added a few tests for `assignments_controller`

Tests to be added for the following controllers : 
- [x] assignments
- [x] collaborations
- [x] group members
- [x] groups
- [x] helpers
- [x] projects
- [x] simulator
- [x] stars
- [x] mailers

### CODE REVIEW

File: spec/support/devise_helper.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+module DeviseRequestSpecHelpers
```
Comment (tachyons) [inline spec/support/devise_helper.rb]: This is already available in devise https://github.com/plataformatec/devise/wiki/How-To:-sign-in-and-out-a-user-in-Request-type-specs-(specs-tagged-with-type:-:request) check first method

**Comment by satu0king** [comment]
> Gem ‘rails-controller-testing’

@danllin, reach out to us on gitter https://gitter.im/CircuitVerse for any help!

**Comment by satu0king** [comment]
This is a large change! @vik-y @tachyons Is it possible to review the same?

**Comment by ankitkataria** [comment]
@satu0king The good thing about the pr is that all the additions are for tests and the build ensures that all them are passing. 😅

File: spec/controllers/collaborations_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+require "rails_helper"
+
+describe CollaborationsController, type: :request do
+  before do
+    @author = FactoryBot.create(:user)
+    @project = FactoryBot.create(:project, author: @author)
+  end
+
+  # TODO - fix test
+  # Throws ActionController::UnknownFormat
```
Comment (tachyons) [inline spec/controllers/collaborations_controller_spec.rb]: ?

Comment (ankitkataria) [reply spec/controllers/collaborations_controller_spec.rb]: This was fixed already, forgot to remove the comment. 😓

**Pull request merged**